### PR TITLE
Update core templates to match latest implementation

### PR DESCRIPTION
**Pull request by saeedmb8**

### CODE REVIEW

File: molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/urls.py [hunk 1]
```diff
@@ -22,6 +23,8 @@
     urlpatterns = patterns('', )
 
 urlpatterns += patterns(
+    '',
+    url(r'^smart-css/$', SmartCSS.as_view(), name='smart-css'),
```
Comment (miltontony) [inline molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/urls.py]: You can simplify this by using `TemplateView.as_view(template_name='smart-css.html')`

**Comment by saeedmb8** [comment]
@miltontony @justinvdm ready for review :)

File: molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/requirements.txt [hunk 2]
```diff
@@ -2,3 +2,4 @@ molo.core=={{cookiecutter.molo_version}}
 {% for requirement in cookiecutter.require %}
 {{requirement}}
 {% endfor %}
+django-extensions
```
Comment (miltontony) [inline molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/requirements.txt]: @saeedmb8 what's this needed for

Comment (saeedmb8) [inline molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/requirements.txt]: @miltontony It's not needed, it adds some custom extensions like `./manage.py show_urls`. I thought it's useful for dev, I can remove it If you see it's not necessary :)

Comment (miltontony) [inline molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/requirements.txt]: Ahhh...cool. Perhaps add it to `requirements-dev.txt` instead

File: molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/templates/core/main.html [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
 {% block content %}
     {% homepages %}
-    {% latest_listing_homepage %}
+    {% latest_listing_homepage num_count=10 %}
```
Comment (miltontony) [inline molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/templates/core/main.html]: What's the default value?

**Comment by miltontony** [comment]
:+1: just a few minor questions. Happy for you to merge after reviewing

File: molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/templates/core/main.html (hunk 3 above)
Comment (saeedmb8) [inline molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/templates/core/main.html]: The default value is 5, I just added here for more visibility of this option :)

**Pull request merged**